webapp/src/pages/app/projects/[projectName]/repos/[repoName]/list/[revision].tsx

```diff
@@ -47,9 +50,52 @@ const RepositoryDetailPage = () => {
   );
   const { isOpen, onOpen, onClose } = useDisclosure();
   const [tabIndex, setTabIndex] = useState(0);
+  const dispatch = useAppDispatch();
+
   const handleTabChange = (index: number) => {
     setTabIndex(index);
   };
+
+  const copyToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      await dispatch(createMessage({ title: '', text: 'copied to clipboard', type: 'success' }));
+    } catch (err) {
+      const error: string = ErrorHandler.handle(err);
+      await dispatch(createMessage({ title: 'failed to copy to clipboard', text: error, type: 'error' }));
+    } finally {
+      dispatch(resetState());
+    }
+  };
+
+  const constructApiUrl = (project: string, repo: string, path: string): string => {
+    return `${window.location.origin}/api/v1/projects/${project}/repos/${repo}/contents${path}`;
```

Comment:
It is necessary to provide `revision` as a parameter of the API.
```suggestion
    let apiUrl = `${window.location.origin}/api/v1/projects/${project}/repos/${repo}/contents${path}`;
    if (revision !== 'head') {
      apiUrl += `?revision=${revision}`;
    }
```